### ISSUE TITLE
MANDATORY -- Shape keys English compatibility

### ISSUE THREAD

**Issue by RMDB-GD** [UuuNyaa/blender_mmd_tools#141]
When you translate morphs to english (with copy english names) you can't import any lips motion because mmd tools is binded to japanese morphs names.(AND YOU CANT REVERT BACK!). This makes any kind of facial animation work tedious and painfull.
This issue should have priority

**Comment by lordscales91** [comment]
I don't think this is as critical as you make it seem. If you are going to end up loading a motion data anyway, you don't need to translate the morphs at all.

**Comment by RMDB-GD** [comment]
sure try to morph a model with all japanese names untranslated. Also if u translate  to english you cvan say farewell to all your mmd motion library co s they wont load and u cant revert.

**Comment by fmnijk** [comment]
The names of morphs should be the same as the keyframe channel names stored in VMD. This is the expected behavior.

**Comment by RMDB-GD** [comment]
This is 100% not bcs MMD autotranslate any morphs declared in the PMX file.

**Comment by fmnijk** [comment]
MMD doesn't translate names for you. It just displays the English version of bone names stored in the PMX file. In order to import motions downloaded from the internet, you should not change the Japanese names to English. However, you can change the English names. MMD_tools displays Japanese names in Blender. Perhaps MMD_tools could provide a way to display English version names when importing PMX files.
<img width="284" alt="Snipaste_2024-08-06_03-15-05" src="https://github.com/user-attachments/assets/c66c7a2e-6073-44d6-b2da-13e870f4b7cb">

**Comment by RMDB-GD** [comment]
**"MMD doesn't translate names for you. It just displays the English version of bone names stored in the PMX file. "**

dude again...in mmd the same happens for morphs..i don't fk..g care about bones.mmd tools can revert the translations for bones, but not for morphs

**Comment by fmnijk** [comment]
Translation is experimental. If this is very important to you, you can fix it and create a pull request yourself. Otherwise, you'll have to wait for someone else to fix it. This is the current state of the repo. There are very few people making real contributions.

**Comment by RMDB-GD** [comment]
And i appreciate it.
I really wanted to point it out cos it has been in this state for years.
Cos ppl are lazy and just do preload things instead to dig in animation making...

**Comment by lordscales91** [comment]
> **"MMD doesn't translate names for you. It just displays the English version of bone names stored in the PMX file. "**
> 
> dude again...in mmd the same happens for morphs..i don't fk..g care about bones.mmd tools can revert the translations for bones, but not for morphs. Thaz what the OP says

Blender is not MMD and it is not meant to be a replacement for it. If you are comfortable using MMD just stick to it. You can just use Blender to make any modifications to the model and then export it back and use it in MMD.

**Comment by RMDB-GD** [comment]
Tell me something i don't know...
The issue is there no matter what.
It's a feature available already for bones and morphs, while one works the other does not.

**Comment by rintrint** [comment]
I could try to address this issue. Do you find any other issues related to the translation tool? If so, I can fix them together.

**Comment by RMDB-GD** [comment]
First of all --Thx for your interest on this matter.

At the moment the problem is that the english translation of morphs can't be reverted (like eg bones can do).

Secondly it would be great to have them already translated in shape keys panel too and synced when reversed from Jap to eng and viceversa.

This to increase compatibility with handamade english animation and custom morphs from english mmd to blender.
While doing it ...better consider even pmx models which have only 1 language (like eg only jap or only eng..ecc..)

This option lack in the plugin from 10 years. IT has been a pain to me to custom animate in blender...(thaz why i dropped)

There were script before, but they never worked for me.

Thx again...and good luck for everything.

**Comment by rintrint** [comment]
https://github.com/rintrint/blender_mmd_tools
I am trying to fix it, but I am not sure whether there is a bug or not. Could you help me test it? You can use Blender 4.2 or 4.3.

https://github.com/user-attachments/assets/37b5a28f-1aec-44da-8d47-00b8b5f44e21

**Comment by RMDB-GD** [comment]
I'm not on new Blender i'm on 4.0.
As i remember to reproduce.

load a pmx jap model.
load vmd on said model.
translate morphs to english
..now youre stucked..
you cant revert morphs from eng to jap
and sonce that..you cant use jap vmd anymore..

at least this happened to me from 2.8 blender to 4.0.

**Comment by rintrint** [comment]
I guess you can test it in 4.0 maybe. I didn’t install 4.0, so I didn’t test it. I’m only not busy now. I’m sorry, but if you don’t test it now and later discover an issue, it might take a long time to fix it again. As shown in the video I provided, it works.

For PMX models that have only one language, you might need to provide me with a mapping you often use. I don’t use this function often, so I don’t know an appropriate mapping.

**Comment by RMDB-GD** [comment]
it's not just the option to translate,
but even the ability to revert/switch translation when needed.
It works for bones, but not for shape keys.

**Comment by rintrint** [comment]
I have already tested it.

**Comment by RMDB-GD** [comment]
if it works then it is fixed ...
surely never worked in old vers

**Comment by rintrint** [comment]
I think the name and name_j of morphs should always be the same in MMD tools. Otherwise, implementing a system to differentiate them would require a lot of work and could introduce many bugs.

Instead, someone could add an additional operation script to translate names from English to Japanese.
Additionally, a new type of operation script could be added to switch between names, rather than just copying and overwriting, ensuring that the original names are not lost.

**Comment by RMDB-GD** [comment]
yep..thaz my wish

**Comment by rintrint** [comment]
I found that you can actually switch between names using a combination of copy and restore. Adding a one-step switch script is unnecessary.

**Comment by rintrint** [comment]
I'm planning to add a temporary column so users can edit translations more easily using that temp field.
This should solve your issue and is also simpler to implement on the technical side.

**Comment by RMDB-GD** [comment]
first of all thx to have fixed this..It's now working.
However .... physics should attach to bones undependently from the bones translation..
I explain myself ...when i translated the bones and the rigid  bodies... "some" rigid body got bugged and won't attach to correct "tanslated" bone....bcs ofc it doesn't find the proper name...reversing to jap doesnt fix the problem.

..However..

_Imo rigid bodies should be able to attach to correct bones ..ignoring the bone name or for example storing someewhere the correct bone's name ..so to be attached properly indipendently from the bones names/translation. i guess using the same temp trick you used for bones_..

Edit: in fact i've checked it rite now....rigid bodies won't attach to correct bone if the bones is translated to english ....
rigid bodies should ignore the translation...and use the model's main language as standard..

i guess the same apply for joints.

without this...it makes the translation pointless..
in the end..i guess translation should modify only the user/visual interface parts and not the model itself

edit:i tested it again ..and it doesn't happens always...sometimes happen..sometimes don't
